yash-datalake/nested/dataFactory.json

```diff
@@ -0,0 +1,307 @@
+      "metadata": {
+        "description": "Name of the data factory. It Must be globally unique."
+      }
+     },
+     "_artifactsLocation": {
+          "type": "string",
+          "metadata": {
+              "description": "The base URI where artifacts required by this template are located including a trailing '/'"
+          }
+      },
+     "_artifactsLocationSasToken": {
+          "type": "SecureString",
+          "metadata": {
+              "description": "The base URI where artifacts required by this template are located including a trailing '/'"
+          }
+      },
+	  
+    "dataFactoryLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "Location of the data factory. Currently, only East US and East US 2 are supported. "
+      }
+    },
+    "blobContainer": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the blob container in the Azure Storage account."
+      }
+    },
+    "servicePrincipalId": {
+      "type": "string",
+      "metadata": {
+        "description": "The ID of the service principal that has permissions to create HDInsight clusters in your subscription."
+      }
+    },
+    "servicePrincipalKey": {
+      "type": "string",
+      "metadata": {
+        "description": "The access key of the service principal that has permissions to create HDInsight clusters in your subscription."
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "storage account name"
+      }
+    }
+  },
+  "variables": {
+    "yashAzureStorageLinkedServiceName": "LinkedServiceToYashAzureStorage",
+    "clientAzureStorageLinkedServiceName": "LinkedServiceToClientAzureStorage",
+    "onDemandHDInsightLinkedServiceName": "LinkedServiceToOnDemandHDInsight",
+    "inputDatasetName": "DatasetIn",
+    "outputDatasetName": "DatasetOut",
+    "pipeline_name_1": "pipelineForDataMovement",
+    "pipeline_name_2": "pipelineForDataTransformation",
+    "pipeline_name_3": "pipelineForDataTransformation2",
+    "entryFilePath": "Neo4JSparkTransformation-0.0.1-SNAPSHOT-jar-with-dependencies.jar"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.DataFactory/factories",
+      "name": "[parameters('dataFactoryName')]",
+      "apiVersion": "2018-06-01",
+      "location": "[parameters('dataFactoryLocation')]",
+      "resources": [
+        {
+          "apiVersion": "2018-06-01",
+          "name": "[variables('yashAzureStorageLinkedServiceName')]",
+          "type": "linkedservices",
+          "dependsOn": [
+            "[parameters('dataFactoryName')]"
+          ],
+          "properties": {
+            "type": "AzureStorage",
+            "description": "Azure Storage linked service",
+            "typeProperties": {
+              "sasUri": {
+                "type": "SecureString",
+                "value": "[uri(parameters('_artifactsLocation'),concat(parameters('_artifactsLocationSasToken')))]"

```

Comment:
how is this used?  this location is a transient, read-only location

Reply:
Do you want us to use ConnectionString typeProperty here??
But we can't use it as we do not have details on your Storage account key

Please let us know how we need to use it here...

Reply:
How is it used?  I.e. what are you using it for?  What does the sample require?  Do you need a persistent storage account for the workload? (if so, you should provision one)  Is it a prereq? (those can be provisioned as well)?  Is it something else?